configuration/src/main/java/io/airlift/configuration/ConfigurationInspector.java

```diff
@@ -182,8 +182,18 @@ private ConfigAttribute(String attributeName, String propertyName, String defaul
 
             this.attributeName = attributeName;
             this.propertyName = propertyName;
-            this.defaultValue = defaultValue;
-            this.currentValue = currentValue;
+            if (securitySensitive && defaultValue != null) {
+                this.defaultValue = "[REDACTED]";
+            }
+            else {
+                this.defaultValue = defaultValue;
+            }
+            if (securitySensitive && currentValue != null) {
+                this.currentValue = "[REDACTED]";
```

Comment:
I have tested this manually. Not sure if it's possible to add an automated test for this in the current framework.